ApplicationInsightsInternalLogger/src/main/java/com/microsoft/applicationinsights/internal/logger/InternalLogger.java

```diff
@@ -85,33 +103,79 @@ private InternalLogger() {
-        if (!initialized) {
-            try {
-                String loggerLevel = loggerData.get(LOGGER_LEVEL);
-                if (Strings.isNullOrEmpty(loggerLevel)) {
-                    // The user didn't specify the logging level, therefore by default we set that to 'TRACE'
-                    loggingLevel = LoggingLevel.TRACE;
-                    setLoggerOutput(loggerOutputType, loggerData);
-                } else {
-                    try {
-                        // Try to match the user request logging level into our enum.
-                        loggingLevel = LoggingLevel.valueOf(loggerLevel.toUpperCase());
-                        setLoggerOutput(loggerOutputType, loggerData);
-                    } catch (Exception e) {
-                        onInitializationError(String.format("Error: Illegal value '%s' for the SDK internal logger. Logging level is therefore set to 'OFF'", loggerLevel));
-                    }
-                }
+        if (initialized) {
+            return;
+        }
 
-                final String utcId = "UTC";
+        try {
+            String loggerLevel = loggerData.get(LOGGER_LEVEL);
+            if (Strings.isNullOrEmpty(loggerLevel)) {
+                // The user didn't specify the logging level, therefore by default we set that to 'TRACE'
+                loggingLevel = LoggingLevel.TRACE;
+                setLoggerOutput(loggerOutputType, loggerData);
+            } else {
                 try {
-                    dateFormatter.setTimeZone(TimeZone.getTimeZone(utcId));
+                    // Try to match the user request logging level into our enum.
+                    loggingLevel = LoggingLevel.valueOf(loggerLevel.toUpperCase());
+                    setLoggerOutput(loggerOutputType, loggerData);
                 } catch (Exception e) {
-                    new ConsoleLoggerOutput().log(String.format("Failed to find timezone with id='%s'. Using default '%s'", utcId, dateFormatter.getTimeZone().getDisplayName()));
+                    onInitializationError(String.format("Error: Illegal value '%s' for the SDK internal logger. Logging level is therefore set to 'OFF'", loggerLevel));
                 }
-            } finally {
-                initialized = true;
+            }
+
+            final String utcId = "UTC";
+            try {
+                dateFormatter.setTimeZone(TimeZone.getTimeZone(utcId));
+            } catch (Exception e) {
+                new ConsoleLoggerOutput().log(String.format("Failed to find timezone with id='%s'. Using default '%s'", utcId, dateFormatter.getTimeZone().getDisplayName()));
+            }
+        } finally {
+            initialized = true;
+        }
+    }
+
+    public synchronized boolean systemPropertyInitialize() {
+        // if the logger is already initialized, don't init again
+        // if the system properties have already been examined, don't check again
+        if (propsNotFound || initialized) {
```

Comment:
By turning these variables to Atomic we can gurantee that initialization happens only once.

Reply:
See my comment above on it.